nipype/interfaces/fsl/epi.py

```diff
@@ -1232,3 +1232,206 @@ def _run_interface(self, runtime):
+                  "the -f option.")
+        )
+    )
+
+    out_cnr_png = traits.List(
+        File(
+            exists=True,
+            mandatory=False,
+            desc=("Image showing mid-sagittal, -coronal and -axial slices of "
+                  "each b-shell CNR volume. Generated when CNR maps are "
+                  "available.")
+        )
+    )
+
+    out_vdm_png = File(
+        exists=True,
+        mandatory=False,
+        desc=("Image showing mid-sagittal, -coronal and -axial slices of "
+              "the voxel displacement map. Generated when using the -f "
+              "option.")
+    )
+
+    out_residuals = File(
+        exists=True,
+        mandatory=False,
+        desc=("Text file containing the volume-wise mask-averaged squared "
+              "residuals. Generated when residual maps are available.")
+    )
+
+    out_clean_volumes = File(
+        exists=True,
+        mandatory=False,
+        desc=("Text file containing a list of clean volumes, based on "
+              "the eddy squared residuals. To generate a version of the "
+              "pre-processed dataset without outlier volumes, use: "
+              "`fslselectvols -i <eddy_corrected_data> -o "
+              "eddy_corrected_data_clean --vols=vols_no_outliers.txt`")
+    )
+
+
+class EddyQuad(FSLCommand):
+    """
+    Interface for FSL eddy_quad, a tool for generating single subject reports
+    and storing the quality assessment indices for each subject.
+    `User guide <https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/eddyqc/UsersGuide>`_
+
+    Examples
+    --------
+
+    >>> from nipype.interfaces.fsl import EddyQuad
+    >>> quad = EddyQuad()
+    >>> quad.inputs.base_name  = 'eddy_corrected'
+    >>> quad.inputs.idx_file   = 'epi_index.txt'
+    >>> quad.inputs.param_file = 'epi_acqp.txt'
+    >>> quad.inputs.mask_file  = 'epi_mask.nii'
+    >>> quad.inputs.bval_file  = 'bvals.scheme'
+    >>> quad.inputs.bvec_file  = 'bvecs.scheme'
+    >>> quad.inputs.output_dir = 'eddy_corrected.qc'
+    >>> quad.inputs.field      = 'fieldmap_phase_fslprepared.nii'
+    >>> quad.inputs.verbose    = True
+    >>> quad.cmdline # doctest: +ELLIPSIS
+    'eddy_quad eddy_corrected --bvals=bvals.scheme --bvecs=bvecs.scheme \
+--field=fieldmap_phase_fslprepared.nii --eddyIdx=epi_index.txt \
+--mask=epi_mask.nii --output-dir=eddy_corrected.qc --eddyParams=epi_acqp.txt \
+--verbose'
+    >>> res = quad.run() # doctest: +SKIP
+
+    """
+    _cmd = 'eddy_quad'
+    input_spec = EddyQuadInputSpec
+    output_spec = EddyQuadOutputSpec
+
+    def _list_outputs(self):
+        import json
+        outputs = self.output_spec().get()
+        out_dir = os.path.abspath(self.inputs.output_dir)
+        outputs['out_qc_json'] = os.path.join(out_dir, 'qc.json')
+        outputs['out_qc_pdf'] = os.path.join(out_dir, 'qc.pdf')
+
+        with open(outputs['out_qc_json']) as fp:
+            qc = json.load(fp)
```

Comment:
Ah, we'd prefer `glob` to this. The reason is that `_list_outputs()` should be runnable without having already run the interface. With `glob` you'll just get an empty list, while attempting to read a JSON file will actually fail.

If there's no way to predict the outputs entirely without running, that's fine. I was just suggesting that, if you could, that's a little nicer. But we should only depend on `self.inputs` to determine the expected outputs. And I can't remember whether you used `sorted(glob(...))` before, but I would recommend sorting the list so it's consistent across runs. (`glob` does not have to return sorted lists.)